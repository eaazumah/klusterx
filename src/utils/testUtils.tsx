import { configureStore } from '@reduxjs/toolkit'
import '@testing-library/jest-dom'
import { render as rtlRender, renderHook as rtlRenderHook } from '@testing-library/react'
import 'jest-styled-components'
import React from 'react'
import { Provider } from 'react-redux'
import settings from '../redux/settingsSlice'
import { IReduxSate } from '../types/declarations'

interface Options {
  store?: any
  preloadedState?: Partial<IReduxSate>
}

interface OptionsWithRouter {
  store?: any
  preloadedState?: Partial<IReduxSate>
}

const render = (
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: Options = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...options })
}

const renderWithRouter = (
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: OptionsWithRouter = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...options })
}

const renderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  {
    mocks = [],
    preloadedState,
    store = configureStore({
      preloadedState,
      reducer: { settings },
    }),
    ...options
  }: any = {},
) => {
  const Wrapper = ({ children }: { children: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRenderHook(callback, {
    ...options,
    wrapper: Wrapper,
  })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render, renderHook, renderWithRouter }
