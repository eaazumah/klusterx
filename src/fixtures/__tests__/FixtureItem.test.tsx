import { render, screen } from '../../utils/testUtils'
import FixtureItem from '../components/FixtureItem'

describe('FixtureItem', () => {
  test('should render fixture item ', () => {
    const fixture = {
      date: '2021-09-11T12:30:00+00:00',
      homeTeam: 'Team A',
      awayTeam: 'Team B',
      homeScore: 1,
      awayScore: 0,
      upComing: false,
    }

    const { asFragment } = render(<FixtureItem fixture={fixture} />)
    expect(asFragment).toMatchSnapshot()
  })

  test('should render fixture item with data', () => {
    const fixture = {
      date: '2021-09-11T12:30:00+00:00',
      homeTeam: 'Team A',
      awayTeam: 'Team B',
      homeScore: 1,
      awayScore: 0,
      upComing: false,
    }

    const { container } = render(<FixtureItem fixture={fixture} />)
    expect(container).toMatchSnapshot()

    const fixtureItem = container.querySelector('.fixture')
    expect(fixtureItem).toBeInTheDocument()

    const fixtureItemText = fixtureItem?.querySelectorAll('a')
    expect(fixtureItemText).toHaveLength(2)

    const fixtureItemTextHome = fixtureItem?.querySelectorAll('a')[0]
    expect(fixtureItemTextHome).toHaveTextContent('Team A')

    const fixtureItemTextAway = fixtureItem?.querySelectorAll('a')[1]
    expect(fixtureItemTextAway).toHaveTextContent('Team B')

    const fixtureItemScore = fixtureItem?.querySelectorAll('div')[1]
    expect(fixtureItemScore).toHaveTextContent('vs')

    const fixtureItemDate = fixtureItem?.querySelectorAll('div')[3]
    expect(fixtureItemDate).toHaveTextContent('1 - 0')
  })

  test('should render fixture item with data and up coming', () => {
    const fixture = {
      date: '2021-09-11T12:30:00+00:00',
      homeTeam: 'Team A',
      awayTeam: 'Team B',
      homeScore: null,
      awayScore: null,
      upComing: true,
    }

    const { container } = render(<FixtureItem fixture={fixture} />)
    expect(container).toMatchSnapshot()

    const fixtureItem = container.querySelector('.fixture')
    expect(fixtureItem).toBeInTheDocument()

    expect(screen.getByText('Upcoming')).toBeInTheDocument()
  })
})
