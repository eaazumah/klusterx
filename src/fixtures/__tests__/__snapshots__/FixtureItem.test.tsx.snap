// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixtureItem should render fixture item  1`] = `[Function]`;

exports[`FixtureItem should render fixture item with data 1`] = `
.c0 .fixture {
  margin: 15px;
  margin-left: 30px;
  margin-right: 30px;
}

.c0 .badge {
  text-transform: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="fixture mantine-1jggmkl"
    >
      <div
        class="mantine-Group-root mantine-7ftt1"
      >
        <a
          class="mantine-Text-root mantine-bwhnbm"
          href="/fixtures/Team A"
        >
          Team A
        </a>
        <div
          class="mantine-Text-root mantine-4fbajm"
        >
          vs
        </div>
        <a
          class="mantine-Text-root mantine-bwhnbm"
          href="/fixtures/Team B"
        >
          Team B
        </a>
      </div>
      <div
        class="mantine-1gn90f5"
      />
      <div
        class="mantine-Group-root mantine-7ftt1"
      >
        <div
          class="mantine-Text-root mantine-1d564l0"
        >
          1
           - 
          0
        </div>
      </div>
      <div
        class="mantine-1gn90f5"
      />
      <div
        class="mantine-Text-root mantine-1d564l0"
      >
        12:30
      </div>
    </div>
    <div
      class="mantine-Divider-root mantine-Divider-horizontal mantine-urkk9t"
      role="separator"
    />
  </div>
</div>
`;

exports[`FixtureItem should render fixture item with data and up coming 1`] = `
.c0 .fixture {
  margin: 15px;
  margin-left: 30px;
  margin-right: 30px;
}

.c0 .badge {
  text-transform: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="fixture mantine-1jggmkl"
    >
      <div
        class="mantine-Group-root mantine-7ftt1"
      >
        <a
          class="mantine-Text-root mantine-bwhnbm"
          href="/fixtures/Team A"
        >
          Team A
        </a>
        <div
          class="mantine-Text-root mantine-4fbajm"
        >
          vs
        </div>
        <a
          class="mantine-Text-root mantine-bwhnbm"
          href="/fixtures/Team B"
        >
          Team B
        </a>
      </div>
      <div
        class="mantine-1gn90f5"
      />
      <div
        class="mantine-Badge-root badge mantine-4y3x5h"
      >
        <span
          class="mantine-1t45alw mantine-Badge-inner"
        >
          Upcoming
        </span>
      </div>
      <div
        class="mantine-1gn90f5"
      />
      <div
        class="mantine-Text-root mantine-1d564l0"
      >
        12:30
      </div>
    </div>
    <div
      class="mantine-Divider-root mantine-Divider-horizontal mantine-urkk9t"
      role="separator"
    />
  </div>
</div>
`;
